File: Programmers/비밀지도.js
// https://programmers.co.kr/learn/courses/30/lessons/17681

function solution(n, arr1, arr2) {
  const a1 = arr1.map((m) => ("0".repeat(n) + m.toString(2)).slice(-n));
  const a2 = arr2.map((m) => ("0".repeat(n) + m.toString(2)).slice(-n));
  const answer = [];

  for (let i = 0; i < n; i++) {
    let line = "";
    for (let j = 0; j < n; j++) {
      line += parseInt(a1[i][j]) || parseInt(a2[i][j]) ? "#" : " ";
    }
    answer.push(line);
  }

  return answer;
}

const n = 5;
const arr1 = [9, 20, 28, 18, 11];
const arr2 = [30, 1, 21, 17, 28];
// 출력	["#####","# # #", "### #", "# ##", "#####"]

console.log(solution(n, arr1, arr2));

File: Programmers/비밀지도.swift
// https://programmers.co.kr/learn/courses/30/lessons/17681?language=swift
// 비밀지도

func solution(_ n:Int, _ arr1:[Int], _ arr2:[Int]) -> [String] {
    var answer: [String] = []
    
    for i in 0..<n {
        let sum: Int = arr1[i] | arr2[i]
        let binSum: String = String((String(repeating: "0", count: n) + String(sum, radix: 2)).suffix(n))
        let bin2Str: String = binSum.replacingOccurrences(of: "1", with: "#").replacingOccurrences(of: "0", with: " ")
        answer.append(bin2Str)
    }
    return answer
}
